File: examples/x2_fun_nframes.py
#!/usr/bin/env python3
'''
@author  Michele Tomaiuolo - http://www.ce.unipr.it/people/tomamic
@license This software is free - http://www.gnu.org/licenses/gpl.html
'''

import g2d

x, y, dx, count = 40, 40, -2, 0
ARENA_W, ARENA_H = 480, 360

def tick():
    global x, dx, count
    g2d.clear_canvas()
    g2d.draw_image("ball.png", (x, y))
    if "LeftButton" in g2d.current_keys() and count == 0:
        count = 10
        dx = -dx
    if count > 0:
        count -= 1
        x += dx

def main():
    g2d.init_canvas((ARENA_W, ARENA_H))
    g2d.main_loop(tick, 5)  # call tick 5 times/second

main()

File: go/x2_fun_nframes.go
package main

import . "g2d"

var x, y, dx, count = 50, 50, 5, 0
var ARENA_W, ARENA_H, MARGIN = 480, 360, 100
var image = LoadImage("ball.png")

func tick() {
    ClearCanvas()  // Draw background
    DrawImage(image, Point{x, y})  // Draw foreground
    if KeyPressed("Enter") {
        count = 5
    }
    if count > 0 {
        count -= 1
        if x + dx < -MARGIN {
            x = ARENA_W + MARGIN
        }
        if x + dx > ARENA_W + MARGIN {
            x = -MARGIN
        }
        x += dx  // Update ball's position
    }
}

func main() {
    InitCanvas(Point{ARENA_W, ARENA_H})
    SetFrameRate(10)
    MainLoop(tick)  // call tick 10 times/second
}
